libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2746,6 +2770,9 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			annotation.isPure = true;
 	}
 
+	if (!annotation.isPure.set())
+		annotation.isPure = false;
```

Comment:
```suggestion
	if (!annotation.isPure.set())
		annotation.isPure = false;
	else
		solAssert(!annotation.isPure, "");
```

Actually, maybe it would be worthwhile to have a helper on `SetOnce` that does this (i.e. either sets a value or verifies that your new value matches)?

Reply:
Well, in this case it doesn't need to match, it just can't be left unset.

Reply:
Oh, ok then. I did not realize that. To be honest these assignments are pretty hard to review without just reading the whole code and reviewing all possible paths.

Reply:
Yeah, that part was a bit messy, indeed.